cranelift/zkasm_data/generated/memory_i32.zkasm

```diff
@@ -11,162 +11,341 @@ start:
 function_1:
   SP - 1 => SP
   RR :MSTORE(SP)
-  E :MSTORE(SP - 1)
-  B :MSTORE(SP - 2)
-  SP - 2 => SP
+  C :MSTORE(SP - 1)
+  D :MSTORE(SP - 2)
+  E :MSTORE(SP - 3)
+  B :MSTORE(SP - 4)
+  SP - 4 => SP
   0n => B  ;; LoadConst32
   0n => A  ;; LoadConst32
   $ => E :ADD
-  $ => A :MLOAD(MEM:E)
+  ${ (E) % 8 } => A
+  ${ (E) / 8 } => E
+  $ => D :MLOAD(MEM:E + 1)
+  $ => B :MLOAD(MEM:E)
+  ${ B >> (8 * A) } => B
+  ${ (D << (128 - 8 * (A + 1))) | B } => B
+  ${ B & ((1 << 8) - 1) } => B
```

Comment:
This seems alright, although knowing zkasm, I think a quick improvement in steps in the future is going to involve `JNZ` on the result of `% 8` and otherwise execute a single load only. Probably similar with stores as well.